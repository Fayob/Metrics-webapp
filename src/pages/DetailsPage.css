.detail_section {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background: #f6516c;
}