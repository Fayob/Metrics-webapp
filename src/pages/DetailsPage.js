/* eslint-disable max-len */
import React from 'react';
import { useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';
import Navbar from './Navbar';
import Heading from '../component/heading';
import DetailComponent from '../component/detailComponent';

import './DetailsPage.css';

const DetailsPage = () => {
  const { id: target } = useParams();
  const covidDatas = useSelector((state) => state.metrics);
  const targetCountry = covidDatas && covidDatas.covid19Data.find((data) => data.Country === target);

  return (
    <>
      <Navbar navHeading="Countries Covid19 Details" />
      <Heading name={target} confirmedCases={targetCountry && targetCountry.TotalCases} />
      <DetailComponent
        target={target}
        Population={targetCountry && targetCountry.Population}
        TotalRecovered={targetCountry && targetCountry.TotalRecovered}
        TotalTests={targetCountry && targetCountry.TotalTests}
        TotalCases={targetCountry && targetCountry.TotalCases}
        ActiveCases={targetCountry && targetCountry.ActiveCases}
        TotalDeaths={targetCountry && targetCountry.TotalDeaths}
      />
    </>
  );
};

export default DetailsPage;
