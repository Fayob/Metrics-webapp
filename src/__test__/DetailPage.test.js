import React from 'react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
// We're using our own custom render function and not RTL's render.
import renderWithProviders from '../utils/utils';
import DetailsPage from '../pages/DetailsPage';

describe('test for Detail page component', () => {
  test('render Detail page', () => {
    expect(renderWithProviders(<BrowserRouter><DetailsPage /></BrowserRouter>)).toMatchSnapshot();
  });
});
