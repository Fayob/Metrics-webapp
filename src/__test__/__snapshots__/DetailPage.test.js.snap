// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for Detail page component render Detail page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="details"
      >
        <nav
          class="navbar_container"
        >
          <a
            class="arrow_back"
            href="/"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
              />
            </svg>
          </a>
          <div>
            <p>
              Countries Covid19 Details
            </p>
          </div>
          <div
            class="mic_setting"
          >
            <div
              class="mic"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
                />
                <path
                  d="M19 10v2a7 7 0 0 1-14 0v-2"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="19"
                  y2="23"
                />
                <line
                  x1="8"
                  x2="16"
                  y1="23"
                  y2="23"
                />
              </svg>
            </div>
            <div
              class="settings"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
                <path
                  d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
                />
              </svg>
            </div>
          </div>
        </nav>
        <div
          class="heading"
        >
          <div
            class="empty_div"
          />
          <div
            class="heading_div"
          >
            <h2
              class="heading_name"
            />
            <p />
          </div>
        </div>
        <section
          class="detail_section"
        >
          <p
            class="detail_info"
          >
            undefined covid19 detailed Information
          </p>
          <div
            class="detail_container"
          >
            <div
              class="detail_paragraph"
            >
              <p>
                Population:
              </p>
              <div>
                <span />
                <svg
                  class="arrow_right"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="10"
                  />
                  <polyline
                    points="12 16 16 12 12 8"
                  />
                  <line
                    x1="8"
                    x2="16"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </div>
            </div>
            <div
              class="detail_paragraph"
            >
              <p>
                Total Cases:
              </p>
              <div>
                <span />
                <svg
                  class="arrow_right"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="10"
                  />
                  <polyline
                    points="12 16 16 12 12 8"
                  />
                  <line
                    x1="8"
                    x2="16"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </div>
            </div>
            <div
              class="detail_paragraph"
            >
              <p>
                Total Recovery:
              </p>
              <div>
                <span />
                <svg
                  class="arrow_right"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="10"
                  />
                  <polyline
                    points="12 16 16 12 12 8"
                  />
                  <line
                    x1="8"
                    x2="16"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </div>
            </div>
            <div
              class="detail_paragraph"
            >
              <p>
                Total Death:
              </p>
              <div>
                <span />
                <svg
                  class="arrow_right"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="10"
                  />
                  <polyline
                    points="12 16 16 12 12 8"
                  />
                  <line
                    x1="8"
                    x2="16"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </div>
            </div>
            <div
              class="detail_paragraph"
            >
              <p>
                Active Cases:
              </p>
              <div>
                <span />
                <svg
                  class="arrow_right"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="10"
                  />
                  <polyline
                    points="12 16 16 12 12 8"
                  />
                  <line
                    x1="8"
                    x2="16"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </div>
            </div>
            <div
              class="detail_paragraph"
            >
              <p>
                Total Tests:
              </p>
              <div>
                <span />
                <svg
                  class="arrow_right"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="10"
                  />
                  <polyline
                    points="12 16 16 12 12 8"
                  />
                  <line
                    x1="8"
                    x2="16"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </div>
            </div>
          </div>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="details"
    >
      <nav
        class="navbar_container"
      >
        <a
          class="arrow_back"
          href="/"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
            />
          </svg>
        </a>
        <div>
          <p>
            Countries Covid19 Details
          </p>
        </div>
        <div
          class="mic_setting"
        >
          <div
            class="mic"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
              />
              <path
                d="M19 10v2a7 7 0 0 1-14 0v-2"
              />
              <line
                x1="12"
                x2="12"
                y1="19"
                y2="23"
              />
              <line
                x1="8"
                x2="16"
                y1="23"
                y2="23"
              />
            </svg>
          </div>
          <div
            class="settings"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="3"
              />
              <path
                d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
              />
            </svg>
          </div>
        </div>
      </nav>
      <div
        class="heading"
      >
        <div
          class="empty_div"
        />
        <div
          class="heading_div"
        >
          <h2
            class="heading_name"
          />
          <p />
        </div>
      </div>
      <section
        class="detail_section"
      >
        <p
          class="detail_info"
        >
          undefined covid19 detailed Information
        </p>
        <div
          class="detail_container"
        >
          <div
            class="detail_paragraph"
          >
            <p>
              Population:
            </p>
            <div>
              <span />
              <svg
                class="arrow_right"
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <polyline
                  points="12 16 16 12 12 8"
                />
                <line
                  x1="8"
                  x2="16"
                  y1="12"
                  y2="12"
                />
              </svg>
            </div>
          </div>
          <div
            class="detail_paragraph"
          >
            <p>
              Total Cases:
            </p>
            <div>
              <span />
              <svg
                class="arrow_right"
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <polyline
                  points="12 16 16 12 12 8"
                />
                <line
                  x1="8"
                  x2="16"
                  y1="12"
                  y2="12"
                />
              </svg>
            </div>
          </div>
          <div
            class="detail_paragraph"
          >
            <p>
              Total Recovery:
            </p>
            <div>
              <span />
              <svg
                class="arrow_right"
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <polyline
                  points="12 16 16 12 12 8"
                />
                <line
                  x1="8"
                  x2="16"
                  y1="12"
                  y2="12"
                />
              </svg>
            </div>
          </div>
          <div
            class="detail_paragraph"
          >
            <p>
              Total Death:
            </p>
            <div>
              <span />
              <svg
                class="arrow_right"
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <polyline
                  points="12 16 16 12 12 8"
                />
                <line
                  x1="8"
                  x2="16"
                  y1="12"
                  y2="12"
                />
              </svg>
            </div>
          </div>
          <div
            class="detail_paragraph"
          >
            <p>
              Active Cases:
            </p>
            <div>
              <span />
              <svg
                class="arrow_right"
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <polyline
                  points="12 16 16 12 12 8"
                />
                <line
                  x1="8"
                  x2="16"
                  y1="12"
                  y2="12"
                />
              </svg>
            </div>
          </div>
          <div
            class="detail_paragraph"
          >
            <p>
              Total Tests:
            </p>
            <div>
              <span />
              <svg
                class="arrow_right"
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <polyline
                  points="12 16 16 12 12 8"
                />
                <line
                  x1="8"
                  x2="16"
                  y1="12"
                  y2="12"
                />
              </svg>
            </div>
          </div>
        </div>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
